src/cargo/core/compiler/compilation.rs

```diff
@@ -529,3 +529,98 @@ fn target_linker(bcx: &BuildContext<'_, '_>, kind: CompileKind) -> CargoResult<O
     }
     Ok(matching_linker.map(|(_k, linker)| linker.val.clone().resolve_program(bcx.gctx)))
 }
+
```

Comment:
In order to better discover these variables, whether to put this part of the content in a separate file management is better?

Reply:
A simple search for `ENV_VARS_SET_FOR_CRATES` could lead the way, though no objection to this proposal, just lacking of a better module name (naming is hard 😆).